src/data/TotalSupply.ts

```diff
@@ -13,3 +16,31 @@ export function useTotalSupply(token?: Token): TokenAmount | undefined {
 
   return token && totalSupply ? new TokenAmount(token, totalSupply.toString()) : undefined
 }
+
+export function useTotalSupplies(tokens?: (Token | undefined)[]): { [tokenAddress: string]: TokenAmount | undefined } {
```

Comment:
This `useTotalSupplies` hook is required in the `useCalculateLPPortfolio` hook, in order to eventually calculate the total value of a user's LP holdings; we can't use `useTotalSupply` there, since a user's LP holdings may change, and iterating over all LP tokens and calling `useTotalSupply` for each may violate React's invariant of hooks being called in the same order every render.

Reply:
(I also figure this hook could probably be useful elsewhere in the codebase!)